media-source-respec.html

```diff
@@ -1160,15 +1223,67 @@ <h2><dfn>MediaSourceHandle</dfn> Object</h2>
+        object has been transferred, it cannot be transferred again.</p>
+
       <pre class="idl">
-[Exposed=(Window,DedicatedWorker)]
+[Transferable, Exposed=(Window,DedicatedWorker)]
 interface MediaSourceHandle {};</pre>
 
-      <section><h2>Transfer</h2>
-        The {{MediaSourceHandle}} [=transfer steps=] and [=transfer-receiving steps=] require the implementation to
-        maintain an implicit internal slot referencing the underlying {{MediaSource}} to enable [=attaching to a media
-        element=] using {{HTMLMediaElement/srcObject}} and consequent setup of an attachment's [=cross-context
-        communication model=].
+      <section><h2><dfn>Transfer</dfn></h2>
+        <p>The {{MediaSourceHandle}} [=transfer steps=] and [=transfer-receiving steps=] require the implementation to
+          maintain an implicit internal slot referencing the underlying {{MediaSource}} to enable [=attaching to a media
+          element=] using {{HTMLMediaElement/srcObject}} and consequent setup of an attachment's [=cross-context
+          communication model=].</p>
+
+        <p class="note">
+          Implementors should be aware that assumption of "move" semantics implied by {{Transferable}} is not always
+          reality. For example, extensions or internal implementations of postMessage using broadcast may cause
+          unintended multiple recipients of a transferred {{MediaSourceHandle}}. For this reason, implementations are
+          guided to not resolve which potential clone of a transferred {{MediaSourceHandle}} is still valid for
+          attachment until and unless any handle for the underlying {{MediaSource}} object is used in the asynchronous
+          portion of the media element's resource selection algorithm. This is similar to the existing behavior for
+          attachment via [=MediaSource object URLs=], which can be cloned easily, where such a URL is valid for at most
+          one attachment start (across all of its potentially many clones).
+        </p>
+
+        <p>
+          Implementations MUST support at most one attachment (load) via {{HTMLMediaElement/srcObject}} ever for the
+          {{MediaSource}} object underlying a {{MediaSourceHandle}}, regardless of potential cloning of the
+          {{MediaSourceHandle}} due to varying implementations of {{Transferable}}.
+          <p class="note">See [=attaching to a media element=] for how this is enforced during the asynchronous portion
+            of the media element's resource selection algorithm.</p>
+        </p>
+
+        <p>{{MediaSourceHandle}} is only exposed on {{Window}} and {{DedicatedWorkerGlobalScope}} contexts, and cannot
+          successfully transfer between different agent clusters. Transfer of a {{MediaSourceHandle}} object can only
+          succeed within the same agent cluster.
+          <p class="note">For example, transfer of a {{MediaSourceHandle}} object from either a {{Window}} or
```

Comment:
I'm not familiar with the term "agent cluster". Is there a reference that could be added here ?

Reply:
Adding a reference is a good idea. Agent Cluster is [defined](https://tc39.es/ecma262/#sec-agent-clusters) in ECMAScript, and [referenced](https://html.spec.whatwg.org/#integration-with-the-javascript-agent-cluster-formalism) from HTML.

Reply:
References added in 0842de7